src/adapter/src/coord.rs

```diff
@@ -2677,6 +2680,16 @@ impl Coordinator {
             .flat_map(|item_id| self.catalog.get_entry(item_id).global_ids())
             .collect();
 
+        // Before possibly creating collections, make sure their schemas are correct.
+        //
+        // Across different versions of Materialize the nullability of columns can change based on
+        // updates to our optimizer.
+        self.controller
+            .storage
+            .evolve_nullability_for_bootstrap(storage_metadata, compute_collections)
```

Comment:
Do I understand correctly that?
- This is a write on the read path: starting the read-only replica irrevocably migrates the collection.
- This is fine in the short term because we're only making things more nullable at the data type level, so all the old writes are valid with the new schema.
- This is fine in the longer term because it should be a noop at the data type level.

Reply:
Exactly! Also, we only migrate collections for Materialized Views and Continual Tasks, not Tables or Sources